pallets/settlement/src/lib.rs

```diff
@@ -598,34 +600,36 @@ decl_module! {
         ) -> DispatchResult {
             let did = Identity::<T>::ensure_perms(origin.clone())?;
             with_transaction(|| {
-                let instruction_id = Self::base_add_instruction(did, venue_id, settlement_type, trade_date, value_date, legs)?;
+                let instruction_id = Self::base_add_instruction(did, venue_id, settlement_type, trade_date, value_date, legs.clone())?;
                 let portfolios_set = portfolios.into_iter().collect::<BTreeSet<_>>();
-                Self::affirm_and_maybe_schedule_instruction(origin, instruction_id, portfolios_set.into_iter())
+                let legs_count = legs.iter().filter(|l| portfolios_set.contains(&l.from)).count() as u64;
```

Comment:
I want to save the iteration if we are gonna fail at `base_add_instruction()`.

Reply:
I don't see the advantage of that. We should aim to improve the worst case and that'll happen when `base_add_instruction` succeeds.

Reply:
Might be worth discussing this at the tech-discussion/standup today if for policy syncing purposes.

Reply:
General consensus from meeting was I think to go for lower worst-case weights in a trade-off like this; a key factor in the discussion was nodes sitting idle otherwise.